public/app/features/explore/QueryRows.tsx

```diff
@@ -51,8 +55,12 @@ export const QueryRows = ({ exploreId }: Props) => {
   const onChange = useCallback(
     (newQueries: DataQuery[]) => {
       dispatch(changeQueries({ exploreId, queries: newQueries }));
+
+      if (isCorrelationsEditorMode && !correlationDetails?.dirty && !isLeftPane) {
```

Comment:
Hmmm.... this might be also something to tackle as a follow-up but I feel this component should not know much about correlations. `dispatch(changeCorrelationEditorDetails({ dirty: true }));` could happen inside the `changeQueries` action. WDYT?